src/Klein/Request.php

```diff
@@ -290,7 +290,7 @@ public function params($mask = null, $fill_with_nulls = true)
      *
      * @param string $key       The name of the parameter to return
      * @param mixed $default    The default value of the parameter if it contains no value
-     * @return string
+     * @return string|array|null
```

Comment:
Yea, I totally get that rule. It's just that the actual value of the param may be even more values, such as `int`, `float`, etc, depending on how the request body, route parameters, or other injected parameters, are parsed.

For example I have a JSON handler that decodes JSON request bodies from incoming requests that contain a `Content-Type` header with the value `application/json` and injects the decoded values into the parameters.

So yea, we can't truthfully accurately "predict" what types will be returned here.

There's prior art for this:

- https://github.com/symfony/http-foundation/blob/v3.2.2/Request.php#L715
- https://github.com/php-fig/http-message/blob/1.0.1/src/ServerRequestInterface.php#L225